OpenOversight/tests/test_utils.py

```diff
@@ -94,13 +94,40 @@ def test_filter_by_badge_no(mockdata):
     results = OpenOversight.app.utils.grab_officers(
         {'race': 'Not Sure', 'gender': 'Not Sure', 'rank': 'Not Sure',
          'min_age': 16, 'max_age': 85, 'name': '', 'badge': '12',
-         'dept': department}
+         'dept': department, 'unique_internal_identifier': ''}
     )
     for element in results:
         assignment = element.assignments.first()
         assert '12' in str(assignment.star_no)
 
 
+def test_filter_by_full_unique_internal_identifier_returns_officers(mockdata):
+    department = OpenOversight.app.models.Department.query.first()
+    target_unique_internal_id = OpenOversight.app.models.Officer.query.first().unique_internal_identifier
+    results = OpenOversight.app.utils.grab_officers(
+        {'race': 'Not Sure', 'gender': 'Not Sure', 'rank': 'Not Sure',
+         'min_age': 16, 'max_age': 85, 'name': '', 'badge': '',
+         'dept': department, 'unique_internal_identifier': target_unique_internal_id}
+    )
+    for element in results:
+        returned_unique_internal_id = element.unique_internal_identifier
+        assert returned_unique_internal_id == target_unique_internal_id
+
+
+def test_filter_by_partial_unique_internal_identifier_returns_officers(mockdata):
+    department = OpenOversight.app.models.Department.query.first()
+    identifier = OpenOversight.app.models.Officer.query.first().unique_internal_identifier
+    partial_identifier = identifier[:len(identifier) // 2]
+    results = OpenOversight.app.utils.grab_officers(
+        {'race': 'Not Sure', 'gender': 'Not Sure', 'rank': 'Not Sure',
+         'min_age': 16, 'max_age': 85, 'name': '', 'badge': '',
+         'dept': department, 'unique_internal_identifier': partial_identifier}
+    )
+    for element in results:
+        returned_identifier = element.unique_internal_identifier
+        assert returned_identifier == identifier
```

Comment:
tests look :100:, very clear!

two other tests that could be added would be covering the following cases:
* department does have `uii` set, the `div` with `id="uii-question"` is present
* department does not have `uii` set, the `div` with `id="uii-question"` is not present

Reply:
Sure thing, I'll implement those tests.  Are there any similar tests that you'd suggest I look at as a model to follow?  I initially tried writing the below
```python3
def test_find_officer_cannot_see_uii_question_for_depts_without_uiis(mockdata, client, session):
    with current_app.test_request_context():
        dept_without_uii = Department.query.filter_by(uii=None).one_or_none()
        form = FindOfficerForm(dept=dept_without_uii)

        rv = client.post(
            url_for('main.get_officer'),
            data=form.data,
            follow_redirects=True
        )
        assert not "Do you know any part of the Officer's" in rv.data.decode('utf-8')
```
The test doesn't work, because even when the form gets filled out with a department that has no uii, `rv.data.decode('utf-8')` includes the uii question.  I think this is a reflection of how the template always contains that question, but the `complaint.js` script hides the question depending on the department id. 
(Also, I might try testing for if the `<div id=uii-question>` is hidden, rather than testing for the question wording.  I realize testing for specific phrases can get brittle.)

Reply:
Does the javascript hide the `#uii-question` div based on the department selection? If so you'll probably need to use selenium to test whether the div gets hidden since that uses an actual browser, and the response you store in the `rv` variable above is just the basic HTML, without the javascript being evaluated.

Reply:
Thank you!  I wrote two tests using selenium and have them passing.  I'll push the changes tonight.